### PR TITLE
Mingw fix

### LINKED ISSUE #29: Win32EventLogAppender.dll build Fails in local machine with WindRes
When running the build with windres I get this:
```
[INFO] --- maven-antrun-plugin:1.8:run (add-dll) @ win32dll ---
[INFO] Executing tasks

main:

clean:
   [delete] Deleting directory C:\Users\claudiow\git\log4jna\win32dll\target

ver:
     [copy] Copying 1 file to C:\Users\claudiow\git\log4jna\win32dll\src

windres:
    [mkdir] Created dir: C:\Users\claudiow\git\log4jna\win32dll\target\obj
     [exec] windres: can't open file `MSG00001.bin': No such file or directory
     [exec] Result: 1

compile:

windres-echo:
     [echo] windres could not be found or failed, Win32EventLogAppender.dll build skipped.

build:
[INFO] Executed tasks
```
The **MSG00001.bin** is not found, and that is OK as it is deleted by the `clean `task.

I soupouse thet it should be generated by the `windres`task

### PR DESCRIPTION
**Pull request by claudiow**
This fixes #29 and updated the build to auto-detect the operating system. 
The default for windows machines is VC as this is what we use in CI.
The default on Linux is MinGW.

Please review the README.md in win32dll folder and let me know if you find anything, English is not my mother tong and more often than not I make mistakes.

### CODE REVIEW

File: win32dll/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Win32EventLogAppender.dll 
+
+The Win32EventLogAppender.dll can build either with Microsoft Visual C or GCC using MinGW on UN*X or Windows.
```
Comment (dblock) [inline win32dll/README.md]: build -> be built with ...

File: win32dll/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Building in Winods
+### Building with MS Visual C
+Execute the appropriate `vcvars[32 | 64 | all]` or launch a developer command line. See 
+[this](https://msdn.microsoft.com/en-us/library/f2ccy3wt.aspx) for more on Visual C command line.
+
+To build use your IDE or run `mvn clean compile` or any goal that will include compile like `deploy` or `package`.
+
+### Building with MinGW and gcc
+Install MinGW and add `<min-gw-home>\bin' to your `PATH` variable.
```
Comment (dblock) [inline win32dll/README.md]: Quotes are off, after bin wrong quote.

**Comment by dblock** [comment]
I made a few minor textual comments, feel free to address them and merge. 

Generally I think we want to move this to the CONTRIBUTING document or maybe make it a BUILDING.md document at the root, nobody would find this otherwise. And it should be a little more straightforward, less explanation more "do this" kind of stuff. But either way any documentation is good.

**Comment by claudiow** [comment]
Typos fixed. I'm working on documentation at par with the site, 
I agree that it is hidden, my plan is to rewrite  README.md

For the time been I added a note and a link in [CONTRIBUTING.md](https://github.com/claudiow/log4jna/blob/mingw-fix/CONTRIBUTING.md) below the build parragraph.

**Pull request merged**